fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerMultiTargetTest.java

```diff
@@ -775,4 +777,153 @@ public void benchMarkMultiTarget() {
         System.out.printf("%d indexes, %d records. Single build took %d milliSeconds, MultiIndex took %d%n",
                 indexes.size(), numRecords, endSingle - startSingle, endMulti - startMulti);
     }
+
+    @Test
+    void testMultiTargetIndexingBlocker() {
+        // block indexing, then release
+        final FDBStoreTimer timer = new FDBStoreTimer();
+        final int numRecords = 107;
+        final int chunkSize  = 17;
+
+        List<Index> indexes = new ArrayList<>();
+        indexes.add(new Index("indexD", new GroupingKeyExpression(EmptyKeyExpression.EMPTY, 0), IndexTypes.COUNT));
+        indexes.add(new Index("indexA", field("num_value_2"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS));
+        indexes.add(new Index("indexB", field("num_value_3_indexed"), IndexTypes.VALUE));
+        indexes.add(new Index("indexC", field("num_value_unique"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS));
+
+        openSimpleMetaData();
+        populateData(numRecords);
+
+        FDBRecordStoreTestBase.RecordMetaDataHook hook = allIndexesHook(indexes);
+        openSimpleMetaData(hook);
+        disableAll(indexes);
+
+        // 1. partly build multi
+        buildIndexAndCrashHalfway(chunkSize, 2, timer, newIndexerBuilder()
+                .setTargetIndexes(indexes));
+
+        // 2. Block indexing
+        FDBRecordStoreTestBase.RecordMetaDataHook localHook = allIndexesHook(indexes);
+        openSimpleMetaData(localHook);
+        String luka = "my name is Luka";
+        try (OnlineIndexer indexer = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setSubspace(subspace)
+                .setTargetIndexes(indexes)
+                .build()) {
+            indexer.indexingStamp(OnlineIndexer.IndexingStampOperation.BLOCK, luka, 10);
+        }
+
+        // 3. query, ensure blocked
+        openSimpleMetaData(hook);
+        try (OnlineIndexer indexer = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setSubspace(subspace)
+                .setTargetIndexes(indexes)
+                .build()) {
+            final AbstractMap<String, IndexBuildProto.IndexBuildIndexingStamp> stampMap =
```

Comment:
Perhaps it would also be good to query before blocking the index build

Reply:
✅